### PR TITLE
[windows][msbuild] Copy entire directory when the native reference is a framework

### PR DESCRIPTION
**Pull request by spouliot**
The entire `.framework` directory needs to be copied back to Windows
when a native reference is a [xc]framework. Otherwise important files
will be missing and the app bundle will be unusable.

Fix https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1339824

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

File: tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/MTouchTaskTests.cs [hunk 1]
```diff
@@ -9,7 +10,7 @@
 
 namespace Xamarin.iOS.Tasks
 {
-	class CustomMTouchTask : MTouchTaskBase
+	class CustomMTouchTask : MTouch
```
Comment (tj-devel709) [inline tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/MTouchTaskTests.cs:13]: Why this change?

**Review by tj-devel709 — APPROVED** [review]

File: tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/MTouchTaskTests.cs (hunk 1 above)
Comment (spouliot) [reply tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/MTouchTaskTests.cs:13]: The `GetAdditionalItemsToBeCopied` I modified (and added tests to) is not part of `MTouchTaskBase`, only from the`MTouch` subclass.

@emaf will correct me if needed :)

`*Base` tasks generally have all the logic required for running on the Mac.

The version running on Windows subclass them to add the behaviour required to remote the execution.

The version running on Mac also subclass `*Base` but it's empty (on purpose). That makes the msbuild target files identical (same names for tasks).

**Comment by spouliot** [comment]
```
Package source with Name: local-dotnet-feed added successfully.
/Users/builder/azdo/_work/1/s/xamarin-macios/dotnet/package/common.csproj(52,3): error : Failed to resolve SDK 'Microsoft.DotNet.SharedFramework.Sdk/6.0.0-beta.21212.6'. Package restore was successful but a package with the ID of "Microsoft.DotNet.SharedFramework.Sdk" was not installed.
/Users/builder/azdo/_work/1/s/xamarin-macios/dotnet/package/common.csproj(52,31): error MSB4236: The SDK 'Microsoft.DotNet.SharedFramework.Sdk/6.0.0-beta.21212.6' specified could not be found. [/Users/builder/azdo/_work/1/s/xamarin-macios/dotnet/package/Microsoft.tvOS.Runtime.tvossimulator-x64/package.csproj]
/Users/builder/azdo/_work/1/s/xamarin-macios/dotnet/package/common.csproj(52,3): error : Failed to resolve SDK 'Microsoft.DotNet.SharedFramework.Sdk/6.0.0-beta.21212.6'. Package restore was successful but a package with the ID of "Microsoft.DotNet.SharedFramework.Sdk" was not installed.
/Users/builder/azdo/_work/1/s/xamarin-macios/dotnet/package/common.csproj(52,31): error MSB4236: The SDK 'Microsoft.DotNet.SharedFramework.Sdk/6.0.0-beta.21212.6' specified could not be found. [/Users/builder/azdo/_work/1/s/xamarin-macios/dotnet/package/Microsoft.iOS.Runtime.ios-arm64/package.csproj]
/Users/builder/azdo/_work/1/s/xamarin-macios/dotnet/package/common.csproj(52,3): error : Failed to resolve SDK 'Microsoft.DotNet.SharedFramework.Sdk/6.0.0-beta.21212.6'. Package restore was successful but a package with the ID of "Microsoft.DotNet.SharedFramework.Sdk" was not installed.
/Users/builder/azdo/_work/1/s/xamarin-macios/dotnet/package/common.csproj(52,31): error MSB4236: The SDK 'Microsoft.DotNet.SharedFramework.Sdk/6.0.0-beta.21212.6' specified could not be found. [/Users/builder/azdo/_work/1/s/xamarin-macios/dotnet/package/Microsoft.iOS.Runtime.iossimulator-x86/package.csproj]
make[1]: *** [nupkgs/Microsoft.tvOS.Runtime.tvossimulator-x64.14.5.100-ci.pr.gh11868.990+sha.9d25d1f2a.nupkg] Error 1
make[1]: *** Waiting for unfinished jobs....
```

File: tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/MTouchTaskTests.cs (hunk 1 above)
Comment (emaf) [reply tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/MTouchTaskTests.cs:13]: That's correct, but there are no more empty subclasses now :). Because of .NET6 we "unified" both task implementations (Windows and Mac) on this repo, but what Sebastien said is still valid:
- Base classes have the core logic of the task
- The unified subclass contains code that enables running the task remotely (from Windows), but if it's being executed from a Mac it directly executes the base class https://github.com/xamarin/xamarin-macios/blob/main/msbuild/Xamarin.iOS.Tasks/Tasks/MTouch.cs#L16-L17. MTouch is one of the most complex tasks we have, ACTool is another good example because it's way simpler https://github.com/xamarin/xamarin-macios/blob/main/msbuild/Xamarin.iOS.Tasks/Tasks/ACTool.cs#L10-L13. ACTool does not implement ITaskCallback so there's no implementation for `GetAdditionalItemsToBeCopied` in that case, that's just for customizing what files need to be copied when executing remotely.

**Review by emaf — APPROVED** [review]

File: tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/MTouchTaskTests.cs (hunk 1 above)
Comment (tj-devel709) [reply tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/MTouchTaskTests.cs:13]: Ah thank you so much @spouliot and @emaf for those very detailed responses! Makes more sense now!

**Comment by dotMorten** [comment]
@spouliot I might still be hitting this with VS17.1preview and .NET MAUI: https://developercommunity.visualstudio.com/t/remote-deploy-to-iphone-fails-with-packageinspecti/1600912

**Comment by spouliot** [comment]
@dotMorten I'm no longer working for Microsoft. You better file a bug report and link it to this PR. FWIW it'a more likely a similar issue then the same one since there tests were added, in the same PR, for the original problem.

**Comment by dotMorten** [comment]
@spouliot Thanks and best of luck in your new adventure. Bug already logged (link right above)
